docker/docker_image.go

```diff
@@ -103,3 +107,50 @@ func GetRepositoryTags(ctx context.Context, sys *types.SystemContext, ref types.
 	}
 	return tags, nil
 }
+
+// GetDigest returns the image's digest. (This is a
+// backward-compatible shim method which calls the module-level
+// GetDigest)
+func (i *Image) GetDigest(ctx context.Context) (*digest.Digest, error) {
+	return GetDigest(ctx, i.src.c.sys, i.src.logicalRef)
+}
+
+// GetDigest returns the image's digest
+func GetDigest(ctx context.Context, sys *types.SystemContext, ref types.ImageReference) (*digest.Digest, error) {
```

Comment:
Typically we return `digest.Digest` by value (except for the `instanceDigest` values, which use a pointer to represent an optional value), and just return `""` in the error cases.